content/docs/1.10.0/advanced-resources/rebuilding/_index.md

```diff
@@ -1,4 +1,186 @@
+            - If **YES**, Longhorn skips the corresponding snapshot data transfer, which reduces the CPU usage, disk IO, network IO, and rebuilding time. [Fast Replica Rebuilding](#fast-replica-rebuilding)
+            - If **NO**, Longhorn calculates and compares block-by-block checksums for the snapshot file between the target replica and the source replica (The checksum is computed using SHA-512 method). If there is a mismatch, the corresponding data block will be synced. [Delta Replica Rebuilding](#delta-replica-rebuilding)
+
+### Full Replica Rebuilding
+
+If the replica is unrecoverable or replenished, Longhorn will synchronize the whole data from the healthy replica. Reconstruct the replica by sending the full snapshot chain.
+
+Full replica rebuilding will waste network bandwidth and generate numerous disk WRITE operations on the target replica node, but it is the only way to rebuild a replica if there is no data on the target node.
+
+### Delta Replica Rebuilding
+
+- This is available for failed replica reusage only, and there is an existing snapshot file (with the same name) in the failed replica data directory
+- There is no checksum/checksum file for the snapshot file. It's like a fallback of fast rebuilding
+
+- **Pros**:
+  - Reduce network bandwidth consumption.
+- **Cons**:
+  - Increased CPU overhead.
+  - Rebuilding time is influenced by CPU performance.
+
+### Fast Replica Rebuilding
+
+Fast rebuilding is enabled when:
+
+- The fast replica building setting is enabled:
+  - `fast-replica-rebuild-enabled: true`
+- Snapshot checksum files are created:
+  - `snapshot-data-integrity`: `enabled`, and a scheduled job will calculate checksums of all snapshots in a configured interval or
+  - `snapshot-data-integrity-immediate-check-after-snapshot-creation: true`, calculate the snapshot checksum after the snapshot is created.
+
+  However, the checksum calculation negatively impacts the storage performance.
+
+6. Skip checking unchanged snapshots if the checksums are the same in the checksum files of snapshots.
+7. If the value of snapshot checksum files between the target replica and the source replica does not exist or is not identical, it will go to the delta rebuilding flow.
+
+- **Pros**:
+  - Minimize network bandwidth consumption.
+  - Minimize disk IO.
+- **Cons**:
+  - Calculating the snapshot checksum can be time-consuming.
+  - The timing of checksum calculation is unpredictable. It can be triggered even if the node is experiencing heavy IO load.
+
+The snapshot checksum is computed:
+
+- Immediately after the snapshot is created with the `snapshot-data-integrity-immediate-check-after-snapshot-creation` setting enabled.
+- Regularly (by default: 7 days) if the `snapshot-data-integrity` setting is `enabled`.
+
+For more details, [Fast Replica Rebuilding](./fast-replica-rebuilding).
```

Comment:
Seems redundant. Please consolidate it into the start of the section
```
- The fast replica building setting is enabled:
  - `fast-replica-rebuild-enabled: true`
- Snapshot checksum files are created:
  - `snapshot-data-integrity`: `enabled`, and a scheduled job will calculate checksums of all snapshots in a configured interval or
  - ....
```

Reply:
Merged. Thanks.